app/services/chart_builder.rb

```diff
@@ -1,35 +1,63 @@
 class ChartBuilder
-  def initialize(benchmark_runs)
+  
+  # @columns is an array that looks like [{ name: "benchmark1", data: [1.1, 1.2] }]
+  # @categories is a an array of version hashes: 
+  # [
+  #   {
+  #     version: "0",
+  #     environment: "ruby 2.2.0dev"
+  #   },
+  #   {
+  #     version: "0",
+  #     environment: "ruby 2.2.0dev"
+  #   }
+  # ]
+  attr_accessor :columns, :categories
+
+  # the metric this benchmark measures, which looks like:
+  # {
+  #   name: "Memory used",
+  #   unit: "Bytes"
+  # }
+  attr_reader :benchmark_result_type
+
+  # `cache_read` looks like 
+  # { datasets: [{ name: "benchmark1", data: [1.1, 1.2] }], versions: [version_hash, version_hash] }
+  def self.construct_from_cache(cache_read, benchmark_result_type)
+    chart_builder = ChartBuilder.new([], benchmark_result_type)
```

Comment:
You don't need to reference the class name from itself. You can do `self.class.new` here